### PR TITLE
JSHint cleanup

### PR DESCRIPTION
**Pull request by mc10**
Various QOL cleanups across the board.

EDIT: This is now good to go, but this definitely needs another pair of eyes (at the very least).

### CODE REVIEW

File: lib/core/utils.js [hunk 1]
```diff
@@ -3,6 +3,29 @@ $.fn.safeHtml = function(string) {
 	else return $(this).html(RESUtils.sanitizeHTML(string));
 };
 
+if (typeof Array.prototype.find !== 'function') {
+	Array.prototype.find = function(predicate) {
```
Comment (andytuba) [inline lib/core/utils.js]: :+1:

File: lib/modules/selectedEntry.js [hunk 2]
```diff
@@ -168,27 +166,26 @@ addModule('selectedEntry', function(module, moduleID) {
 		}
 	}
 
+	function scrollTo(selected, last, options) {
+		if (!selected) return;
+		options = $.extend(true, {}, {
+				makeVisible: selected.entry,
```
Comment (erikdesjardins) [inline lib/modules/selectedEntry.js]: Very minor but the comma on this line and the empty object can be removed.

**Comment by erikdesjardins** [comment]
I didn't look extra hard, but everything else seems to look good.

File: lib/modules/hosts/imgur.js [hunk 3]
```diff
@@ -38,20 +38,16 @@ modules['showImages'].siteModules['imgur'] = {
...
 						def.resolve(json.data);
 						return;
 					}
 				} catch (error) {
-					// debugger;
 					def.reject(error);
 					return;
 				}
-				// debugger;
 				def.reject(response);
```
Comment (erikdesjardins) [inline lib/modules/hosts/imgur.js]: This function could do with even more cleanup: this line is unreachable, and the return statements are somewhat redundant.

**Comment by mc10** [comment]
Done, thanks for the suggestions.

File: lib/core/storage.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
-var RESStorage = RESStorage || {};
+/* exported RESStorage, RESLoadResourceAsText */
```
Comment (andytuba) [inline lib/core/storage.js]: Some day we'll corral all of these into `RES.thing...`

File: lib/core/utils.js [hunk 5]
```diff
@@ -653,15 +699,19 @@ RESUtils.getHeaderOffset = function() {
...
-		if ((obj[i]) && (obj[i].value === value)) {
-			obj[i].selected = true;
-		}
+	// Only call Array.prototype.slice.call if obj is array-like
+	// e.g. getElementsByTagName, but *not* getElementById
+	if (obj.length !== undefined) {
+		obj = Array.prototype.slice.call(obj);
+	} else {
+		obj = [obj];
 	}
```
Comment (andytuba) [inline lib/core/utils.js]: We should probably replace this with `$(obj).val(value)`

**Comment by andytuba** [comment]
note to self: pick up again at accountSwitcher

File: lib/core/utils.js (hunk 5 above)
Comment (mc10) [inline lib/core/utils.js]: That's a good point, I'll do that.

**Comment by mc10** [comment]
Rebased to the latest changes on `master`.

**Comment by andytuba** [comment]
phew, what a day.  Hope all those `!== null`'s work out as expected.

**Pull request merged**